services/motion/builtin/builtin.go

```diff
@@ -178,15 +189,36 @@ func (ms *builtIn) MoveOnMap(
 	slamName resource.Name,
 	extra map[string]interface{},
 ) (bool, error) {
-	operation.CancelOtherWithLabel(ctx, "motion-service")
+	operation.CancelOtherWithLabel(ctx, motionOpId)
 
 	// get the SLAM Service from the slamName
-	slamService, ok := ms.slamServices[slamName]
+	local, ok := ms.localizers[slamName]
 	if !ok {
 		return false, resource.DependencyNotFoundError(slamName)
 	}
 	ms.logger.Warn("This feature is currently experimental and does not support obstacle avoidance with SLAM maps yet")
 
+	// assert localizer as a slam service and get map limits
+	slamSvc, ok := local.(localizer.SLAMLocalizer)
+	if !ok {
+		return false, fmt.Errorf("cannot assert local of type %T as slam service", local)
+	}
+
+	// gets the extents of the SLAM map
+	data, err := slam.GetPointCloudMapFull(ctx, slamSvc)
+	if err != nil {
+		return false, err
+	}
+	dims, err := pointcloud.GetPCDMetaData(bytes.NewReader(data))
+	if err != nil {
+		return false, err
+	}
+	limits := []referenceframe.Limit{
```

Comment:
I think this should also take into account the start and goal

Reply:
I think I disagree.
My reasoning is that the slam map shows up exactly what we can plan on.

It would not make sense to select a destination which has not yet been mapped, so in that way the destination cannot be used to "inflate"/change the existing limits.
Similarly I cannot imagine a scenario where our current position would change the existing limits.

I am unfamiliar with the guts of how moveonmap does its magic, but maybe there is a scenario of moving from one edge of the map to another - perhaps it is wise to add a buffer to the limits. Something like 2x the diameter of the moving base?
let me know your thoughts.

Reply:
I'm ok with keeping this as is for now, we can always change it later if we need to.  Was mostly thinking of keeping things consistent between the two methods